src/main/java/apaintus/controllers/ToolBarController.java

```diff
@@ -124,7 +124,7 @@ public ActiveTool getActiveTool() {
     }
 
     private void setColorPicker(ColorPicker colorPicker, String colorValue) {
-        if(!colorValue.isEmpty()) {
+        if(colorValue != null &&  !colorValue.isEmpty()) {
```

Comment:
is the color value is null it should throw an error since it is "not possible" to have a null color

Reply:
Pourquoi il y a une possibilite d'avoir null @boscoa ?

Reply:
je sais pas pourquoi le downvote garni, clairement, bosco a mis ce check parce que cela arrivait. Si jamais cela arrive, cela devrait throw une erreur et avertir que cela n'est pas possible, non? En plus de fixer la raison pourquoi cela arrive en ce moment aussi

Reply:
Je n'ai jamais eu une String null dans Java jusqu'à maintenant, mais la SelectionBox provoquait un null sur la color quand elle était sélectionnée

Reply:
S'il le met pas, il va pogner un NullPointerException donc voila ton erreur. C'est plus important de verifier pourquoi null pour se retrouver a cet endroit la. Possiblement utiliser un Optional<> si necessaire.

Reply:
Le truc, c'est dans la factory, je ne build pas avec le strokeColor sur la selectionBox car il s'agirait d'une information non pertinente et son parent Shape set à null sa valeur. En utilisant un Optional, faudrait faire une condition pour vérifier qu'il est présent de toute façon, non?

Reply:
Ouin... On a maybe un refactor a faire sur les objets qui peuvent etre dessines. Si on commence a avoir plus que des "shape", faudrait reviser comment on fait

Reply:
Sinon, on pourrait ajouter dans les constructeurs des shapes des conditions de null... Du style : this.width = (shapeAttributes.getWidth() == null) ? 0 : shapeAttributes.getWidth();
À ce moment, on s'assure qu'aucune shape n'aura un élément set à null.